packages/proto-rpc/src/ClientTransport.ts

```diff
@@ -0,0 +1,138 @@
+import {
+    ClientStreamingCall,
+    ServerStreamingCall,
+    DuplexStreamingCall,
+    Deferred,
+    RpcTransport,
+    MethodInfo,
+    RpcError,
+    RpcOptions,
+    RpcMetadata,
+    RpcStatus,
+    UnaryCall,
+    mergeRpcOptions
+} from '@protobuf-ts/runtime-rpc'
+import { v4 } from 'uuid'
+import { TODO } from './types'
+import { RpcMessage } from './proto/ProtoRpc'
+import EventEmitter = require('events')
+import { Logger } from './Logger'
+
+export enum Event {
+    RPC_REQUEST = 'streamr:dht-transport:request-new'
+}
+
+export interface ClientTransport {
+    on(event: Event.RPC_REQUEST, listener: (deferredPromises: DeferredPromises, rpcMessage: RpcMessage, options: ProtoRpcOptions) => void): this
+}
+
+export interface DeferredPromises {
+    header: Deferred<RpcMetadata>,
+    message: Deferred<object>,
+    status: Deferred<RpcStatus>,
+    trailer: Deferred<RpcMetadata>,
+    messageParser: (bytes: Uint8Array) => object
+}
+
+export interface ProtoRpcOptions extends RpcOptions {
+    notification?: boolean
+}
+
+const logger = new Logger(module)
+
+export class ClientTransport extends EventEmitter implements RpcTransport {
+    static objectCount = 0
+    private objectId = 1
+    private stopped = false
+    protected readonly defaultOptions: TODO
+
+    constructor(defaultTimeout?: number) {
+        super()
+        this.objectId = ClientTransport.objectCount
+        ClientTransport.objectCount++
+        
+        this.defaultOptions = {
+            timeout: defaultTimeout || 5000,
+            clientId: this.objectId
+        }
+    }
+
+    mergeOptions(options?: Partial<ProtoRpcOptions>): RpcOptions {
+        return mergeRpcOptions(this.defaultOptions, options)
+    }
+
+    createRequestHeaders(method: MethodInfo, notification?: boolean): any {
+        return {
+            method: method.localName,
+            request: 'request',
+            notification: notification ? 'notification' : undefined
+        }
+    }
+
+    unary<I extends object, O extends object>(method: MethodInfo<I, O>, input: I, options: ProtoRpcOptions): UnaryCall<I, O> {
+        const
+            requestBody = method.I.toBinary(input),
+            defHeader = new Deferred<RpcMetadata>(),
+            defMessage = new Deferred<O>(),
+            defStatus = new Deferred<RpcStatus>(),
+            defTrailer = new Deferred<RpcMetadata>()
```

Comment:
While perfectly valid code, I would use separate variable declaration to conform to existing style.
```suggestion
        const requestBody = method.I.toBinary(input)
        const defHeader = new Deferred<RpcMetadata>()
        const defMessage = new Deferred<O>()
        const defStatus = new Deferred<RpcStatus>()
        const defTrailer = new Deferred<RpcMetadata>()
```